src/aod_sketch_pg_functions.c

```diff
@@ -118,89 +125,43 @@ Datum pg_aod_sketch_add_item(PG_FUNCTION_ARGS) {
   }
 
   if (PG_ARGISNULL(0)) {
-    lg_k = PG_NARGS() > 3 ? PG_GETARG_INT32(3) : 0;
+    stateptr = palloc(sizeof(struct aod_agg_state));
+    stateptr->type = MUTABLE_SKETCH;
+    stateptr->lg_k = PG_NARGS() > 3 ? PG_GETARG_INT32(3) : 0;
+    stateptr->num_values = arr_len;
     p = PG_NARGS() > 4 ? PG_GETARG_FLOAT4(4) : 1;
-    if (lg_k) {
-      sketchptr = p ? aod_sketch_new_lgk_p(arr_len, lg_k, p) : aod_sketch_new_lgk(arr_len, lg_k);
+    if (stateptr->lg_k) {
+      stateptr->ptr = p ? aod_sketch_new_lgk_p(arr_len, stateptr->lg_k, p) : aod_sketch_new_lgk(arr_len, stateptr->lg_k);
```

Comment:
I think this will always take the first branch unless p is explicitly set to 0?

Reply:
yeah, I think this is a mistake, but harmless. I will fix